linux/desktop_webview_auth_plugin.cc

```diff
@@ -0,0 +1,175 @@
+#include "include/desktop_webview_auth/desktop_webview_auth_plugin.h"
+
+#include <flutter_linux/flutter_linux.h>
+#include <gtk/gtk.h>
+#include <glib.h>
+#include <cstring>
+
+#include <webkit2/webkit2.h>
+
+#define DESKTOP_WEBVIEW_AUTH_PLUGIN(obj)                                     \
+  (G_TYPE_CHECK_INSTANCE_CAST((obj), desktop_webview_auth_plugin_get_type(), \
+                              DesktopWebviewAuthPlugin))
+
+DesktopWebviewAuthPlugin *plugin;
```

Comment:
there should be no need for a global variable. the plugin instance can be passed around as "user data".